src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -40,23 +40,29 @@ public void init() {
   }
 
   /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
 
-    String user = request.getParameter("user");
-
-    if (user == null || user.equals("")) {
+    // Get logged-in user and other user
+    String loggedInUser = null;
```

Comment:
I would suggest using `Optional<String>` for `loggedInuser` and `otherUser`.

Reply:
I'm confused about this. I don't think that `loggedInUser` or `otherUser` should be optional, so would using `Optional<String>` imply that I can not put pass in one of them? Because if that were the case, then you wouldn't be able to see any messages at all.

Reply:
The `Optional` type doesn't necessarily mean that it is valid for the object to be absent, just as the use of a nullable type doesn't necessarily mean that `null` is a legal/permissible value.

Rather, the use of `Optional` simply draws attention to the fact that absence is a possibility and is something that needs to be taken into consideration. It is common practice in many Java code bases to assume that anything which is not using the `Optional` type or that is not annotated with the `@Nullable` type annotation will be non-null (and thus checking for null is unnecessary).

When you use:

     String user = ...

It makes it less obvious that code below must test that `user != null`.

If you, instead, write:

     Optional<String> user = ...

... then it makes it more obvious that the code below must check that the value is present.

You can always assign to a pure, non-optional type once the check has been applied as in:

      Optional<String> userOrAbsent = ...
      if (userOrAbsent.orElse("").isEmpty()) {
         // ... generate an error ...
        return;
      }
      String user = userOrAbsent.get();